support/chez/support.ss

```diff
@@ -439,6 +439,51 @@
     (condition-signal read-cv)
     the-val))
 
+(define (blodwen-channel-get-non-blocking ty chan)
+  (if (mutex-acquire (channel-read-mut chan) #f)
+    (let* ([val-box  (channel-val-box  chan)]
+           [read-box (channel-read-box chan)]
+           [read-cv  (channel-read-cv  chan)]
+           [the-val  (unbox val-box)]
+          )
+      (if (null? the-val)
+          (begin
+           (mutex-release (channel-read-mut chan))
+           '())
+          (begin
+            (set-box! val-box '())
+            (set-box! read-box #t)
+            (mutex-release (channel-read-mut chan))
+            (condition-signal read-cv)
+            the-val)
+      ))
+  '()))
+
+(define (blodwen-channel-get-with-timeout ty chan seconds)
+  (let loop ([start-time (current-time)]
+            )
+    (let ([elapsed (time-difference (current-time) start-time)])
+      (if (time>=? elapsed (make-time 'time-duration 0 seconds))
+          '()
+          (if (mutex-acquire (channel-read-mut chan) #f)
+              (let* ([val-box  (channel-val-box chan)]
+                     [the-val  (unbox val-box)])
+                (if (null? the-val)
+                    (begin
+                      (mutex-release (channel-read-mut chan))
+                      (sleep (make-time 'time-duration 1000000 0))
+                      (loop start-time))
+                    (let* ([read-box (channel-read-box chan)]
+                           [read-cv  (channel-read-cv chan)])
+                      (set-box! val-box '())
+                      (set-box! read-box #t)
+                      (mutex-release (channel-read-mut chan))
+                      (condition-signal read-cv)
+                      the-val)))
+              (begin
+                (sleep (make-time 'time-duration 1000000 0))
+                (loop start-time)))))))
+
```

Comment:
Someone familiar with scheme internals really needs to look this over carefully.

Reply:
Yes, would definitely appreciate some additional eyes on this.